src/renderer/components/SaveChangesButton.js

```diff
@@ -17,154 +17,95 @@
-    marginTop: -12,
-    marginLeft: -12,
-  },
-  icon: {
-    marginRight: -16,
-    zIndex: 1,
-  },
-  disabled: {
-    backgroundColor: theme.palette.action.disabled,
-  },
-  fab: {
-    position: "fixed",
-    justifyContent: "flex-end",
-    bottom: 0,
-    left: theme.spacing(4),
-    zIndex: theme.zIndex.drawer + 1,
-  },
-});
+const SaveChangesButton = (props) => {
+  const [inProgress, setInProgress] = React.useState(false);
+  const [success, setSuccess] = React.useState(false);
 
-class SaveChangesButton extends React.Component {
-  state = {
-    inProgress: false,
-    success: false,
-  };
-
-  componentWillUnmount() {
-    clearTimeout(this.timer);
-  }
+  const { classes, successMessage } = props;
 
-  handleButtonClick = async (event) => {
-    this.setState(
-      {
-        inProgress: true,
-      },
-      async () => {
-        await this.props.onClick(event);
-        this.setState({
-          success: true,
-          inProgress: false,
-        });
-        this.timer = setTimeout(() => {
-          this.setState({ success: false });
-        }, 2000);
-      }
-    );
+  const handleButtonClick = async (event) => {
+    setInProgress(true);
+    console.log("about to do aysnc sutff");
+    console.log(" about to call  the onlick handler");
+    await props.onClick(event);
+    console.log("Got back from click");
+    setSuccess(true);
+    setInProgress(false);
+    setTimeout(() => {
+      console.log("running the timeout callback ");
+      setSuccess(false);
+      console.log("finishied timeout callback");
+    }, 2000);
   };
 
-  render() {
-    const { inProgress, success } = this.state;
-    const { classes, successMessage } = this.props;
-    const buttonClassname = classNames({
-      [classes.buttonSuccess]: success,
-    });
-
-    const textPart = !this.props.floating && (
-      <div className={classes.wrapper}>
-        <Button
-          variant="contained"
-          color="primary"
-          className={buttonClassname}
-          disabled={inProgress || (this.props.disabled && !success)}
-          onClick={this.handleButtonClick}
-        >
-          {success
-            ? successMessage || i18n.t("components.save.success")
-            : this.props.children}
-        </Button>
-      </div>
-    );
+  const textPart = !props.floating && (
+    <Box className={classes.wrapper}>
+      <Button
+        variant="contained"
+        disabled={inProgress || (props.disabled && !success)}
+        onClick={handleButtonClick}
+        color={success ? "success" : "primary"}
+      >
+        {success
+          ? successMessage || i18n.t("components.save.success")
+          : props.children}
+      </Button>
+    </Box>
+  );
 
-    const icon = this.props.icon || <SaveAltIcon />;
+  const icon = props.icon || <SaveAltIcon />;
 
-    const OptionalTooltip = (props) => {
-      if (this.props.floating) {
-        return <Tooltip title={this.props.children}>{props.children}</Tooltip>;
```

Comment:
This is one place where the component breaks: in the old code, we had `props` (local to `OptionalTooltip`) and `this.props` (from `SaveChangesButton`). We need to treat those separately.

We probably want to `const { children } = props` outside of `OptionalTooltip`, and use `children` and `props` children in there. Or just rename the `props` argument of `OptionalTooltip`.